EssentialFeed/EssentialFeed/Image Comment Feature/ImageCommentLoader.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  ImageCommentLoader.swift
+//  EssentialFeed
+//
+//  Created by Ángel Vázquez on 17/03/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public protocol ImageCommentLoaderTask {
+	func cancel()
+}
+
+public protocol ImageCommentLoader {
+	typealias Result = Swift.Result<[ImageComment], Error>
+	
+	func load(from url: URL, completion: @escaping (Result) -> Void) -> ImageCommentLoaderTask
```

Comment:
`URL` is an implementation detail of the `Remote` loader. Not *every* implementation should need a URL.

The protocol should not expose implementation details. Otherwise, it's not a good abstraction.

The URL should be injected into the concrete Remote loader implementation via initializer injection instead.

```suggestion
	func load(completion: @escaping (Result) -> Void) -> ImageCommentLoaderTask
```